File: arrays/lastStoneWeight.py
# Last Stone Weight - Leetcode

def lastStoneWeight(self, stones):
        stones.sort(reverse = True)
        
        while(len(stones) > 1):
            
            if stones[0] == stones[1]:
                del stones[0:2]
                
                
            else:
                most = stones[0]
                sec_most = stones[1]
                del stones[0:2]
                stones.append((most - sec_most))
                stones.sort(reverse = True)
                
        if stones:
            return stones[0]
        else:
            return 0

File: heap/lastStoneWeight.java
// Leetcode - 1046 - Last Stone Weight
public int lastStoneWeight(int[] stones) {
    PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());
    for(int val : stones) maxHeap.add(val);
    while(maxHeap.size() > 1) {
        int stone1 = maxHeap.poll();
        int stone2 = maxHeap.poll();
        if(stone1 != stone2) {
            maxHeap.add(stone1 - stone2);
        }
    }
    return maxHeap.isEmpty() ? 0 : maxHeap.peek();
}